lib/http/retriable/performer.rb

```diff
@@ -168,11 +168,20 @@ def out_of_retries_error(request, response, exception)
       end
 
       def build_retry_proc(exception_classes, retry_statuses)
+        retry_statuses = [retry_statuses].flatten
+
         ->(_req, err, res, _i) {
           if err
             exception_classes.any? { |e| err.is_a?(e) }
           else
-            retry_statuses.include?(res.status.to_i)
+            response_status = res.status.to_i
+            retry_statuses.any? do |matcher|
+              case matcher
+              when Range then matcher.include?(response_status)
```

Comment:
`Range#cover?` is more efficient:

```suggestion
              when Range then matcher.cover?(response_status)
```